src/Microsoft.OData.Client/Serialization/HttpWebRequestMessage.cs

```diff
@@ -63,149 +63,143 @@ public class HttpWebRequestMessage : DataServiceClientRequestMessage
         private bool sendingRequest2Fired;
 #endif
 
-        #endregion // Private Fields
-
-#if PORTABLELIB
         /// <summary>
+        /// Constructor for HttpWebRequestMessage.
         /// Initializes the <see cref="HttpWebRequestMessage"/> class.
         /// </summary>
-        static HttpWebRequestMessage()
+        public HttpWebRequestMessage(DataServiceClientRequestMessageArgs args) 
+            : base(args.ActualMethod)
         {
-            IsRunningOnSilverlight = typeof(System.Windows.Input.ICommand).GetAssembly().GetType("System.Net.Browser.WebRequestCreator", false) != null;
+            _requestMessage = new HttpRequestMessage();
+            _messageStream = new MemoryStream();
+            _handler = new HttpClientHandler();
+            _client = new HttpClient(_handler, disposeHandler: true);
+            _contentHeaderValueCache = new Dictionary<string, string>();
 
-            PropertyInfo pi = typeof(HttpWebRequest).GetProperty("ContentLength");
-            if (pi != null)
-            {
-                ContentLengthMethodSetter = pi.GetSetMethod();
-            }
+            _effectiveHttpMethod = args.Method;

```

Comment:
@ElizabethOkerio Curious... There's `args.ActualMethod` passed to the base class and then there's `args.Method` used here. How/when do they differ? Preflight?

Reply:
difference is in post tunneling.